assets/app/scripts/services/data.js

```diff
@@ -708,7 +706,7 @@ angular.module('openshiftConsole')
 
   DataService.prototype._urlForType = function(type, id, context, isWebsocket, params) {
```

Comment:
actually, factor that into a `normalizeType` function, and call it for all the Data functions that take a type
```
function normalizeType(type, caller) {
   if (!type) return type;
   var lower = type.toLowerCase();
   if (type != lower) {
     // log warning with type and caller name
   }
   return lower;
}